stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -68,6 +68,12 @@
    * exception is thrown by an implementation of {@code onError} no further calls to any method are
    * allowed.
    *
+   * <p>{@code t} should be a {@link io.grpc.StatusException} or {@link
+   * io.grpc.StatusRuntimeException}, but other {@code Throwable} types are possible. Callers should
```

Comment:
As an overrider you are told you don't care too much about non-`Status*Exception`. You still handle it, but you don't have to worry about it as much, which is why `Status.fromThrowable()` works so well.

As a caller, you should always pass a `Status*Exception`. But if you fail to, it's 'okay'.